### PR TITLE
React showrecipe

### PR DESCRIPTION
**Pull request by tomlouismurphy**
### Ticket Number:
6

### Describe The Problem Being Solved:

This is an early react page for ticket #6, in which a user or visitor can visit recipe pages. It includes components for serving size, duration, difficulty level, and individual tags. This is still in a very early stage of development and design and hasn't been connected to Rails endpoints, but I thought it was good to have on here just so that we could start being able to share components.

screen shot 2018-10-29 at 6 31 15 pm
screen shot 2018-10-29 at 6 31 38 pm
screen shot 2018-10-29 at 6 31 49 pm

### Checklist For Submitter

* [x ] I have documented all new functionality
* [x ] I have screenshotted new UI changes and attached them to this PR

### Checklist For Reviewer
* [ ] Code formatting/static analysis: is the code formatted properly? Does a linter/other form of static analysis reveal any issues? Does the code have appropriately low cyclomatic complexity?
* [ ] Code architecture: concerns are separated, code follows single-responsibility principle, code utilizes OOP, DRY code.
* [ ] Coding best practices: code avoids hard-coded variables, inefficient computations, reinventing the wheel when using frameworks. Code is well-commented and generally readable.
* [ ] Non-functional requirements: Code has accompanying tests (if using TDD). Programmer can easily explain decisions to reviewers.

### CODE REVIEW

**Comment by tomlouismurphy** [comment]
Kept working on this. The comment section now is operational, albeit not hooked up into a database - just using dummy data and comments only stay on the page until reload, not saved in any way

File: frontend/public/index.html [hunk 1]
```diff
@@ -36,5 +38,8 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
+    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
```
Comment (EvanGrillo) [inline frontend/public/index.html]: generally it's recommended to not use jquery library in React. worth looking into

File: frontend/public/index.html [hunk 2]
```diff
@@ -36,5 +38,8 @@
...
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/js/bootstrap.min.js" integrity="sha384-JZR6Spejh4U02d8jOt6vLEHfe/JQGiRRSQQxSfFWpi1MquVdAyjUar5+76PVCmYl" crossorigin="anonymous"></script>
```
Comment (EvanGrillo) [inline frontend/public/index.html]: different bootstrap version from Ryan's PR

File: frontend/public/index.html (hunk 1 above)
Comment (tomlouismurphy) [reply frontend/public/index.html]: good catch yep. haven't been using jquery so far so I can remove this

**Comment by tomlouismurphy** [comment]
Notes on what is still left to include (apart from hooking everything up into the back end): need to add optional delete and edit tags on the header that pop up for admins and user who posted recipe; need to fully optimize for mobile devices; need to ensure there aren't any merge conflicts now that Zach's merged the super-admin work into master; making a few css/bootstrap fixes on the header and the tag bar

File: frontend/src/RecipeBase/RecipeBase.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import './RecipeBase.css';
+import {MainImage} from './MainImage/MainImage.js';
+import {SpecsBar} from '../SpecsBar/SpecsBar.js';
+import {TagsBar} from '../TagsBar/TagsBar.js';
+import {MainBody} from './MainBody/MainBody.js';
+import {IngredientList} from './IngredientList/IngredientList.js';
+import {Directions} from './Directions/Directions.js';
+import {CommentSection} from './CommentSection/CommentSection.js';
+
+class Base extends Component {
```
Comment (EvanGrillo) [inline frontend/src/RecipeBase/RecipeBase.js]: May be best change Base to RecipeBase. As we all discussed, some of these components might be best baked back into the RecipeBase markup unless they're reusable pieces that will own state

File: frontend/public/index.html (hunk 2 above)
Comment (tomlouismurphy) [reply frontend/public/index.html]: changed to same CDN as Ryan's

File: frontend/src/RecipeBase/RecipeBase.js (hunk 3 above)
Comment (tomlouismurphy) [reply frontend/src/RecipeBase/RecipeBase.js]: good call, change made

**Pull request merged**